### PR TITLE
fix regexp

### LINKED ISSUE #2059: discord roles dont get chat filled properly
Issue opened by Discord user: Zamiel#8743

### PR DESCRIPTION
**Pull request by DrAnax**
Closes #2059 

Cannot test (but [code](https://github.com/Hanabi-Live/hanabi-live/blob/bcbe24fe6ef18f1591ffd9aaab68980e3f7bed5b/server/src/chat.go#L93) now fetches the correct user id)

(based on issue screenshot)

### CODE REVIEW

**Comment by DrAnax** [comment]
Rerun the tests

**Comment by Zamiell** [comment]
i think the `&` should have a `*` after it right?
since it isn't in every single mention

**Comment by DrAnax** [comment]
> i think the `&` should have a `*` after it right? since it isn't in every single mention

According to the screenshot, there's only one `&`. Console tests show that the user_id is grabbed correctly.

**Comment by Zamiell** [comment]
correct me if I am wrong, but I think this regex is used to parse both user mentions and role mentions, which are distinct from each other.
the former does not have an ampersand in it, right?

**Comment by DrAnax** [comment]
Right:

```
mentionRegExp = regexp.MustCompile(`&lt;@&*!*(\d+?)&gt;`)
test &lt;@&921716503051632650&gt;
Discord ID: 921716503051632650

test &lt;@921716503051632650&gt;
Discord ID: 921716503051632650
```

vs

```
mentionRegExp = regexp.MustCompile(`&lt;@&!*(\d+?)&gt;`)
test &lt;@&921716503051632650&gt;
Discord ID: 921716503051632650

test &lt;@921716503051632650&gt;
Discord ID: not found
```

File: server/src/chat.go [hunk 1]
```diff
@@ -79,9 +79,9 @@ func chatServerSendPM(s *Session, msg string, room string) {
 }
 
 func chatFillMentions(msg string) string {
-	if discord == nil {
-		return msg
-	}
+	// if discord == nil {
```
Comment (Zamiell) [inline server/src/chat.go:82]: what

Comment (DrAnax) [reply server/src/chat.go:82]: oops, debug, sorry

**Comment by Zamiell** [comment]
good fix, thank you

**Pull request merged**